ancv/web/server.py

```diff
@@ -261,3 +264,124 @@ def server_timing_header(timings: dict[str, timedelta]) -> str:
         f"{name.replace(' ', '-')};dur={duration // timedelta(milliseconds=1)}"
         for name, duration in timings.items()
     )
+
+
+class WebHandler(Runnable):
+    """A handler serving a rendered template loaded from a URL with periodic refresh."""
+
+    def __init__(self, destination: str, refresh_interval: int = 300) -> None:
+        """Initializes the handler.
+
+        Args:
+            destination: The URL to load the JSON Resume from.
+            refresh_interval: How often to refresh the resume in seconds.
+        """
+        self.destination = destination
+        self.refresh_interval = refresh_interval
+        self.cache: Optional[str] = None
```

Comment:
This could be `self.cache: Optional[ResumeSchema] = None`, see below for more!

Reply:
Using self.cache as strong typed `str` in the function split refactoring, because I'm caching the template rendered from `Template.render()` which returns a `str`, done.

Reply:
I made the `self.cache` strong typed `str`, since the implementation is dealing with the rendered template, do you think that's better to change it to `self.cache: Optional[ResumeSchema] = None`, and the functions to deal with that?

Reply:
Hmmm, up to you!

In my mind the cache is for network I/O, and going from `ResumeSchema` to a rendered `str` is so cheap and fast it won't matter unless you're requesting your resume at crazy RPS. I just really hate `str` as a type. It says and guarantees basically nothing. But if `cache` as `str` is a better fit, it's okay!